src/AggregateFunctions/IAggregateFunction.h

```diff
@@ -217,31 +236,90 @@ class IAggregateFunctionHelper : public IAggregateFunction
 
     AddFunc getAddressOfAddFunction() const override { return &addFree; }
 
-    void addBatch(size_t batch_size, AggregateDataPtr * places, size_t place_offset, const IColumn ** columns, Arena * arena) const override
+    void addBatch(
+        size_t batch_size,
+        AggregateDataPtr * places,
+        size_t place_offset,
+        const IColumn ** columns,
+        Arena * arena,
+        size_t num_arguments = 0) const override
     {
-        for (size_t i = 0; i < batch_size; ++i)
-            static_cast<const Derived *>(this)->add(places[i] + place_offset, columns, i, arena);
+        if (num_arguments > 0)
```

Comment:
This looks strange. What if the function already has multiple arguments, but there is no `-If` combinator?

Example: `corr`.

Reply:
> This looks strange. What if the function already has multiple arguments, but there is no -If combinator?

It's indeed bad name.  `num_arguments` is served as an `-If` indicator here, and it denotes the pos of `-If` cond, which equals to the nested_func's num_arguments

Reply:
And the only special no-argument agg: `count` has specialized implementation.

Reply:
Let's name it `mask_argument_pos`, it can be ssize_t with -1 as no value.